quickwit-core/src/indexing/split.rs

```diff
@@ -127,17 +132,39 @@ impl Split {
             num_parsing_errors: 0,
             storage,
             metastore,
+            timestamp_field,
         })
     }
 
     /// Add document to the index split.
     pub fn add_document(&mut self, doc: Document) -> anyhow::Result<()> {
-        //TODO: handle time range when docMapper is available
+        let mut computed_time_range = None;
+        if let Some(timestamp_field) = self.timestamp_field {
+            let mut split_time_range = self.metadata.time_range.clone().unwrap_or(Range {
```

Comment:
I would avoid the clone here

Reply:
Ok, My reasoning is that since we are using Range<i64> that is cheap to clone.